x/finality/types/power_table.go

```diff
@@ -140,12 +139,16 @@ func (dc *VotingPowerDistCache) GetInactiveFinalityProviderSet() map[string]*Fin
 	return inactiveFps
 }
 
+// NewFinalityProviderDistInfo loads the FinalityProviderDistInfo based on the fp data
 func NewFinalityProviderDistInfo(fp *bstypes.FinalityProvider) *FinalityProviderDistInfo {
 	return &FinalityProviderDistInfo{
 		BtcPk:          fp.BtcPk,
 		Addr:           sdk.MustAccAddressFromBech32(fp.Addr),
 		Commission:     fp.Commission,
 		TotalBondedSat: 0,
+		IsJailed:       fp.Jailed,
+		IsSlashed:      fp.IsSlashed(),
+		IsTimestamped:  false, // not possible to fill info without tip height
```

Comment:
hmm this assumes the caller will fill this field approprietly. Maybe lets add it in docstring of this function so nobody will forget that ?

Reply:
good idea